test/components/tableTests.ts

```diff
@@ -1,370 +1,472 @@
-    assert.deepEqual(translates[3], [350, 30], "right axis translate");
-    assert.deepEqual(translates[2], [50, 30] , "plot translate");
-    // test the bboxes
-    TestMethods.assertBBoxEquivalence(bboxes[0], [300, 30], "top axis bbox");
-    TestMethods.assertBBoxEquivalence(bboxes[4], [300, 30], "bottom axis bbox");
-    TestMethods.assertBBoxEquivalence(bboxes[1], [50, 340], "left axis bbox");
-    TestMethods.assertBBoxEquivalence(bboxes[3], [50, 340], "right axis bbox");
-    TestMethods.assertBBoxEquivalence(bboxes[2], [300, 340], "plot bbox");
-    svg.remove();
-  });
+    it("throws an Error on trying to add null to the Table", () => {
+      const table = new Plottable.Components.Table();
+      assert.throws(() => table.add(null, 0, 0), "Cannot add null to a table cell");
+    });
 
-  it("table space fixity calculates properly", () => {
-    let tableAndcomponents = generateBasicTable(3, 3);
-    let table = tableAndcomponents.table;
-    let components = tableAndcomponents.components;
-    components.forEach((c) => TestMethods.fixComponentSize(c, 10, 10));
-    assert.isTrue(table.fixedWidth(), "fixed width when all subcomponents fixed width");
-    assert.isTrue(table.fixedHeight(), "fixedHeight when all subcomponents fixed height");
-    TestMethods.fixComponentSize(components[0], null, 10);
-    assert.isFalse(table.fixedWidth(), "width not fixed when some subcomponent width not fixed");
-    assert.isTrue(table.fixedHeight(), "the height is still fixed when some subcomponent width not fixed");
-    TestMethods.fixComponentSize(components[8], 10, null);
-    TestMethods.fixComponentSize(components[0], 10, 10);
-    assert.isTrue(table.fixedWidth(), "width fixed again once no subcomponent width not fixed");
-    assert.isFalse(table.fixedHeight(), "height unfixed now that a subcomponent has unfixed height");
+    it("detaches a Component from its previous location before adding it", () => {
+      const component = new Plottable.Component();
+      const svg = TestMethods.generateSVG();
+      component.renderTo(svg);
+      const table = new Plottable.Components.Table();
+      table.add(component, 0, 0);
+      assert.isFalse((<Node> svg.node()).hasChildNodes(), "Component was detach()-ed");
+      svg.remove();
+    });
   });
 
-  it("table.requestedSpace works properly", () => {
-    let c0 = new Plottable.Component();
-    let c1 = TestMethods.makeFixedSizeComponent(50, 50);
-    let c2 = TestMethods.makeFixedSizeComponent(20, 50);
-    let c3 = TestMethods.makeFixedSizeComponent(20, 20);
-
-    let table = new Plottable.Components.Table([
-      [c0, c1],
-      [c2, c3]
-    ]);
+  describe("removing Components", () => {
+    const c1 = new Plottable.Component();
+    const c2 = new Plottable.Component();
+    const c3 = new Plottable.Component();
+    const c4 = new Plottable.Component();
+    const c5 = new Plottable.Component();
+    const c6 = new Plottable.Component();
```

Comment:
I am suspicious about defining these variables up here.  The problem is that if a test in the suite affects `c1`, then that affects any tests depending on `c1`